### ISSUE TITLE
SimpleDB: ConditionalCheckFailedException

### ISSUE THREAD

**Issue by ogmios-voice** [aws/aws-sdk-java#829]
Add dedicated `ConditionalCheckFailedException` to SimpleDB similar to DynamoDB's `com.amazonaws.services.dynamodbv2.model.ConditionalCheckFailedException`.

Right now in SimpleDB this exception can be detected only by comparing the `statusCode` (int) and `errorCode` (string) fields of the very base `AmazonServiceException`; and the possible errorCode values are defined only in the [API documentation](http://docs.aws.amazon.com/AmazonSimpleDB/latest/DeveloperGuide/APIError.html).

Basically you should have dedicated exception types for all errors that should require special handling in the client.

See also #826.

**Comment by shorea** [comment]
+1, engaging the service team.

**Comment by debora-ito** [comment]
Hi @ogmios-voice, a feature request to the SimpleDB team was created at the time, if they decide to add the exception to their model the SDK will start to see it right away. 

So I'll go ahead and close this. Feel free to reach out if you have any questions or comments.
